mkt/tags/models.py

```diff
@@ -37,25 +37,23 @@ def can_reverse(self):
     def get_url_path(self):
         return reverse('tags.detail', args=[self.tag_text])
 
-    def save_tag(self, addon):
+    def _get_m2m_name(self, obj):
+        model = obj._meta.model_name
+        if model == 'webapp':
+            return 'tags'
+        elif model == 'website':
+            return 'keywords'
```

Comment:
Maybe. You can access the m2m related to the `Tag` model with `Tag._meta.get_all_related_m2m_objects_with_model()`. From the other side, you can use `<Website|Webapp>._meta.many_to_many` and find the one linking to `Tag`.